std/algorithm/comparison.d

```diff
@@ -736,6 +736,27 @@ template equal(alias pred = "a == b")
         {
             return r1 == r2;
         }
+        // if one of the arguments is a string and the other isn't, then auto-decoding
+        // can be avoided if they have the same ElementEncodingType
+        else static if (is(typeof(pred) == string) && pred == "a == b" &&
```

Comment:
The `== string` ensures it will not work with wstring, and wstring's also autodecode.